### PR TITLE
Fix Qwen2.5-VL temporal RoPE for fractional video intervals

### PR DESCRIPTION
**Pull request by yeyeyeping**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48669) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48669)

# What does this PR do?

Fixes premature truncation of fractional video timing in Qwen2.5-VL temporal RoPE.

`get_rope_index` currently converts each `second_per_grid_ts` value to an integer before multiplying by `tokens_per_second`. With a temporal patch size of 2 and an effective sampling rate of 4 FPS, the interval is 0.5 seconds and is truncated to zero. All temporal grids in that video consequently receive the same temporal coordinate. Non-integral intervals above one second also lose their fractional component.

Keep the temporal step floating point and convert the complete per-grid temporal coordinates to `long` in `get_vision_position_ids`. For `tokens_per_second=2`, four grids with 0.5-second intervals now have relative temporal coordinates `[0, 1, 2, 3]` instead of `[0, 0, 0, 0]`. Quantization happens after multiplying by the grid index, so fractional scaled steps such as 1.5 also accumulate correctly.

The Qwen2.5-VL modular source overrides its inherited vision-position helper to keep this behavior change local to Qwen2.5-VL. The generated modeling file is regenerated with the repository converter. Image coordinates and integer-interval behavior are preserved.

This restores the quantization order used by the earlier implementation:
- [Transformers v4.49.0](https://github.com/huggingface/transformers/blob/v4.49.0/src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py#L1533-L1538)
- [Qwen's public fine-tuning implementation](https://github.com/QwenLM/Qwen2.5-VL/blob/main/qwen-vl-finetune/qwenvl/data/rope2d.py#L248-L253)

## Validation

- Regression covers 0.25, 0.5, 0.75, 1.0 and 1.5 seconds, with Python-list and tensor inputs; checks temporal/spatial coordinates, output dtype, and RoPE deltas.
- Running the regression with the two original methods from baseline `4815a0a` produces 8 failing fractional-interval subtests; both integer-interval subtests pass.
- With this change: 10 regression subtests pass, alongside the existing `test_vision_position_ids` and `test_video_forward` tests.
- Ruff lint and format checks pass for all three changed files.
- Modular conversion consistency check passes for Qwen2.5-VL.
- Tested on CPU; no full accelerator suite or downstream accuracy evaluation was run.

Commands (with the checkout's `src` on `PYTHONPATH`):

```bash
python -m pytest tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py -k 'vision_position_ids or video_position_ids_fractional_seconds or test_video_forward' -q
python utils/check_modular_conversion.py --files src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py --check_all --num_workers 1
```

## Before submitting

- [x] Read the repository contribution guidance.
- [x] Update the affected helper's parameter annotation and documentation.
- [x] Add necessary regression tests.
- [ ] Link an upstream issue or discussion, if available.

Prepared for review in the owner's repository. This patch and description were prepared with coding-agent assistance; the upstream template's first-time-contributor declaration has not been asserted. Check the applicable upstream contribution requirements before submitting there.

### CODE REVIEW

File: src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ def get_rope_index(
                     # Only apply temporal scaling for videos; still images have no
                     # temporal dimension to space out (fixes #45325).
                     if modality_type == 2:
-                        time_interval = tokens_per_second * int(next(second_per_grid_ts))
+                        time_interval = tokens_per_second * next(second_per_grid_ts)
                     else:
                         time_interval = 1
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py:1056]: the ref implementation by qwen team used to cast `int` on final input, looks like we are supposed to int (tokens_per_sec * seconds_per_grid) to follow it

Otherwise I dont think we need to "fix" smth if it aligns with ref model-design even if it looks un-intuitive 

https://github.com/huggingface/transformers/blob/f3f6c86582611976e72be054675e2bf0abb5f775/src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py#L1626-L1629

Comment (yeyeyeping) [reply src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py:1056]: Thanks for checking! I agree that we should preserve the reference model behavior. My understanding is based only on comparing Qwen’s official open-source implementation and the earlier Transformers implementation you linked. 

In that code, expanded_range already contains the temporal grid indices, so .long() is applied after multiplying by the grid index, seconds per grid, and tokens per second. This PR follows that quantization order in get_vision_position_ids.

Casting only tokens_per_second * seconds_per_grid would produce different results. For example, with tokens_per_second=2 and seconds_per_grid=0.75, the reference produces [0, 1, 3, 4], while casting the step first produces [0, 1, 2, 3]. The added regression test covers this case.

That comparison is why I thought this was a regression, but I don’t have additional context from the Qwen team. If the current implementation was contributed by them and reflects their intended model behavior, I’ll defer to that and close this PR.

Comment (zucchini-nlp) [reply src/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py:1056]: oke, I checked just now and yes, we had a hidden regression here. Could you also update `get_vision_position_ids` in other models, iirc they are all copied so you'd have to run `make fix-repo`

And i think qwen-omni models also used `seconds_per_grid`, so lets double check if more models need a fix

File: src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py [hunk 2]
```diff
@@ -315,6 +315,58 @@ def __init__(self, config):
...
+        self,
+        start_position: int,
+        grid_thw: list[int, int, int] | torch.Tensor,
+        temp_merge_size: int = 1,
+        spatial_merge_size: int = 1,
+        time_interval: float = 1.0,
+        device: str | torch.device | None = None,
+    ):
+        """
+        Compute 3D positional indices for vision tokens derived from a single image or video input.
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py]: no override in modular, i assume it is supposed to be rounded after all multiplies so lets fix in `qwen2-vl` and run modular for all models

Comment (yeyeyeping) [reply src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py]: Updated in 2fddb94. 

I moved the fix into the shared Qwen2-VL helper, removed the Qwen2.5-VL modular override, and ran make fix-repo to regenerate all dependent models.

Comment (yeyeyeping) [reply src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py]: However, I may have found a separate bug in both Omni processors: video_second_per_grid uses the requested FPS even when max_frames reduces the effective sampling rate.

  I checked Qwen’s original video loader. It recalculates sample_fps after applying frame limits and can return per-video FPS through return_video_kwargs=True:

  https://github.com/QwenLM/Qwen2.5-Omni/blob/3d66ebe50d70292a78a3ee3541d25d44eb74b7d0/qwen-omni-utils/src/qwen_omni_utils/v2_5/vision_process.py#L211-L217

  Qwen2.5-VL correctly uses each video's metadata.sampled_fps to compute second_per_grid_ts:

  https://github.com/huggingface/transformers/blob/2fddb94484785a5576e5de02107905378e0fb792/src/transformers/models/qwen2_5_vl/processing_qwen2_5_vl.py#L138-L149

  Both Omni processors instead reuse the requested FPS. I reproduced this with a 10-second video, requested fps=2, max_frames=8, and temporal_patch_size=2:

  - Effective sampling rate: 0.8 FPS
  - Expected interval: 2.5 seconds
  - Both Omni processors return: 1 second
  - Qwen2.5-VL correctly returns: 2.5 seconds

  This affects temporal positions and audio/video interleaving. The early official example also omits forwarding the sampled FPS, so this may be a longstanding issue rather than a recent regression.

Comment (yeyeyeping) [reply src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py]: I also checked both Omni models for the original premature truncation issue:

  Qwen2.5-Omni keeps the video interval as a float during multiplication and applies .long() only after multiplying the frame index, seconds per grid, and positions per second:

  https://github.com/huggingface/transformers/blob/2fddb94484785a5576e5de02107905378e0fb792/src/transformers/models/qwen2_5_omni/modeling_qwen2_5_omni.py#L407-L409

  Qwen3-Omni-MoE keeps the resulting temporal indices and vision position IDs as floats, so it does not truncate the fractional interval either:

  https://github.com/huggingface/transformers/blob/2fddb94484785a5576e5de02107905378e0fb792/src/transformers/models/qwen3_omni_moe/modeling_qwen3_omni_moe.py#L388-L390

  Neither has the same premature truncation bug.

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py [hunk 3]
```diff
@@ -308,6 +308,38 @@ def test_vision_position_ids(self):
...
+    def test_video_position_ids_fractional_seconds(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+        config.vision_config.tokens_per_second = 2
+        model = Qwen2_5_VLModel(config).to(torch_device)
+        input_ids = torch.tensor(
+            [[config.text_config.pad_token_id] + [config.video_token_id] * 4], device=torch_device
+        )
+        mm_token_type_ids = torch.tensor([[0, 2, 2, 2, 2]], device=torch_device)
+        video_grid_thw = torch.tensor([[4, 1, 1]], device=torch_device)
+
```
Comment (zucchini-nlp) [inline tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:320]: a bit too much of manual input prep but oke, not on you - the prio of a proper VideoTester is rising every day 🙃

**Review by zucchini-nlp — APPROVED** [review]
Thanks! lgtm

File: src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py (hunk 2 above)
Comment (zucchini-nlp) [reply src/transformers/models/qwen2_5_vl/modular_qwen2_5_vl.py]: > video_second_per_grid uses the requested FPS even when max_frames reduces the effective sampling rate.

not really worth it, let's leave it the way it was added initially. These processors haven't been updated in years and don't even follow the recommended processor API, i also doubt it's not used commonly
